legal-api/src/legal_api/services/warnings/business/business_checks/involuntary_dissolution.py

```diff
@@ -0,0 +1,56 @@
+# Copyright © 2024 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Service to check involuntary dissolution for a business."""
+from legal_api.services.involuntary_dissolution import InvoluntaryDissolutionService
+
+from . import BusinessWarningCodes, WarningType
+
+
+def check_business(business: any) -> list:
+    """Check involuntary dissolution for warnings."""
+    result = []
+
+    ar_overdue_warning = {
+        'code': BusinessWarningCodes.MULTIPLE_ANNUAL_REPORTS_NOT_FILED,
+        'message': 'Multiple annual reports not filed.  Eligible for involuntary dissolution.',
+        'warningType': WarningType.NOT_IN_GOOD_STANDING
+    }
+    transition_warning = {
+        'code': BusinessWarningCodes.TRANSITION_NOT_FILED,
+        'message': 'Transition filing not filed.  Eligible for involuntary dissolution.',
+        'warningType': WarningType.NOT_IN_GOOD_STANDING
+    }
+
+    eligibility, details = InvoluntaryDissolutionService.check_business_eligibility(business.identifier, True)
+    if eligibility:
+        if details.ar_overdue:
+            result.append(ar_overdue_warning)
+        if details.transition_overdue:
+            result.append(transition_warning)
+    elif business.in_dissolution:
+        _, dis_details = InvoluntaryDissolutionService.check_business_eligibility(business.identifier, False)
+        if dis_details.ar_overdue:
+            result.append(ar_overdue_warning)
+        if dis_details.transition_overdue:
+            result.append(transition_warning)
+        batch_processing = business.batch_processing.one_or_none()
```

Comment:
I think this needs to be a more precise query.  The batch tables could be used for other things in the future.

I think you should create a function in `InvoluntaryDissolutionsService` that does what `business.in_dissolution` essentially does.

``` python
@staticmethod
    def get_in_dissolution_batch_processing(business_id: int):
        """Fetch the BatchProcessing record for a business that is in the process of involuntary dissolution."""
        return db.session.query(BatchProcessing, Batch).\
            filter(BatchProcessing.business_id == business_id).\
            filter(BatchProcessing.status.notin_([BatchProcessing.BatchProcessingStatus.COMPLETED,
                                                  BatchProcessing.BatchProcessingStatus.WITHDRAWN])). \
            filter(Batch.id == BatchProcessing.batch_id).\
            filter(Batch.status != Batch.BatchStatus.COMPLETED).\
            filter(Batch.batch_type == Batch.BatchType.INVOLUNTARY_DISSOLUTION).\
            one_or_none()
```